tree-sitter/tree-sitter-c-sharp
# Add syntax highlight generator

This PR adds a small C# project to generate highlight test files. 

Steps to generate the test file:

- Add a sample `.cs` file in the `test/highlight` folder (with no highlight comments).
- run the generator with in the repo root:
```
./tools/highlight-test-generator/run-generator [ABSOLUTE_PATH_TO_FILE.cs]
``` 

The generator is quite slow. It performs the following:
- identifies tokens in the input based on character types;
- adds comments in the format `// <- identifier` or `//   ^ identifier`, where the `identifier` is unique;
- runs the following in an infinite loop until it finds changes:
  - runs `tree-sitter test`;
  - parses the highlighting errors (`Failure - row: 333, column: 66, expected highlight 'a47286482_21', actual highlights: 'punctuation.bracket'`) in the output;
  - finds the unique identifier (`a47286482_21`) in the input file, and replaces it with the actual highlight (`punctuation.bracket`).
